pennylane/workflow/qnode.py

```diff
@@ -753,10 +756,40 @@ def circuit(x):
 
         original_init_args.update(kwargs)
         updated_qn = QNode(**original_init_args)
+
         # pylint: disable=protected-access
         updated_qn._transform_program = qml.transforms.core.TransformProgram(self.transform_program)
         return updated_qn
 
+    def update_shots(self, shots: Union[int, Shots]) -> "QNode":
+        """Update the number of shots used by the QNode.
+
+        Args:
+            shots (int or Shots): The new number of shots to use.
+
+        Returns:
+            qnode (QNode): new QNode with updated shots
+        """
+
+        # Create a copy of the current QNode
+        updated_qn = copy.copy(self)
+
+        # Update the shots attribute directly
+        # pylint: disable=protected-access
+        updated_qn._set_shots(shots)
+
+        return updated_qn
+
+    def _set_shots(self, shots: Union[int, Shots]) -> None:
+        """Set the number of shots used by the QNode.
+
+        Args:
+            shots (int or Shots): The new number of shots to use.
+        """
+
+        self._shots = Shots(shots)
+        self._shots_override_device = bool(self._shots != self.device.shots)
```

Comment:
I don't understand the purpose of this `_shots_override_device` flag. Is there ever a scenario where we used `set_shots` to change `self._shots`, but still use `device.shots` in execution? Doesn't `self._shots` **always** take precedence over `device.shots`?

Reply:
We designed this for raising certain warnings e.g. in `__call__`. Although right now it's only used one place. When the `device.shots` is deprecated in the future, it will be totally useless.

Reply:
NVM found it